core/src/test/java/org/apache/calcite/plan/RelWriterTest.java

```diff
@@ -740,6 +740,53 @@ private static Fixture relFn(Function<RelBuilder, RelNode> relFn) {
         .assertThatPlan(isLinux(expected));
   }
 
+  /** Test case for
+   * <a href="https://issues.apache.org/jira/browse/CALCITE-6703">[CALCITE-6703]
+   * RelJson cannot handle timestamps prior to 1970-01-25 20:31:23.648</a>. */
+  @Test void testJsonToRexForTimestamp() {
+      final String timestampRepresentedAsInt = "{\n"
+        + "  \"literal\": 2129400000,\n"
+        + "  \"type\": {\n"
+        + "    \"type\": \"TIMESTAMP\",\n"
+        + "    \"nullable\": false\n"
+        + "  }\n"
+        + "}\n";
+      final String timestampRepresentedAsLong = "{\n"
+        + "  \"literal\": 3129400000,\n"
```

Comment:
not easy to check that this value corresponds to the date below.

Reply:
Yeah I also wasn't sure about this.. the point I was trying to illustrate is that one case is below Integer.MAX_VALUE and the other is above, any suggestions?

Reply:
putting this in a comment will help a bit - the purpose of this choice is clearer
but if you have validated this using other external means, like another DB, you can also mention it in a comment.

Reply:
Ok i've added some comments